packages/block-editor/src/components/block-title/use-block-display-title.js

```diff
@@ -54,7 +55,11 @@ export default function useBlockDisplayTitle( {
 			const label = getBlockLabel( blockType, attributes, context );
 			// If the label is defined we prioritize it over a possible block variation title match.
 			if ( label !== blockType.title ) {
-				return label;
+				// Users can enter and save a label made of only spaces e.g. for
+				// navigation links. In that case we provide a fallback label.
+				// We do want to trim leading and trailing spaces anyways.
+				const trimmedLabel = label.trim();
+				return trimmedLabel === '' ? __( 'Untitled' ) : trimmedLabel;
```

Comment:
I think this should be handled in `getBlockLabel`, which should return a fallback (block title) when only spaces are provided.

Then, we don't need to handle this case by case.